### PR TITLE
Fix autocomplete issues

### PR DESCRIPTION
**Pull request by blittle**

### CODE REVIEW

File: src/select.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import {findIndex} from 'lodash';
 
-let searchString;
+let searchString = "";
```
Comment (blittle) [inline src/select.js]: If not initialized as an empty string, below the first search will be `"undefinedU"` or whatever character you press.

File: src/select.js [hunk 2]
```diff
@@ -167,12 +167,11 @@ const CanopySelect = React.createClass({
 		var i = this.getIndexFromString(searchString);
 
 		if(i > -1) {
-			this.setState({
-				selectedIndex: i,
-				//			dialogDisplayed: true
-			});
+			this.selectItem(i);
```
Comment (blittle) [inline src/select.js]: If the value has been highlighted, it has been changed, calling `selectItem` will triggers an `onChange` event.

File: src/select.js [hunk 3]
```diff
@@ -167,12 +167,11 @@ const CanopySelect = React.createClass({
...
 		}
 
+		clearTimeout(keyTimeout);
```
Comment (blittle) [inline src/select.js]: If the timeout isn't cleared each time we come into this method, then the search string will always reset after 1 second, instead we want them to be able to keep typing. Only reset if they stop typing for longer than a second in between each key.

File: lib/select.js [hunk 4]
```diff
@@ -199,15 +199,15 @@ var CanopySelect = _react2['default'].createClass({
 
 	highlightByText: function highlightByText(charCode) {
 		searchString += String.fromCharCode(charCode);
+		console.log(searchString);
```
Comment (dckesler) [inline lib/select.js]: Probably don't want this

**Comment by dckesler** [comment]
Other than the console.log :cactus:

**Pull request merged**